Source/WebCore/Modules/ShapeDetection/Implementation/Cocoa/BarcodeDetectorImplementation.mm

```diff
@@ -23,28 +23,205 @@
  * THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include "config.h"
-#include "BarcodeDetectorImplementation.h"
+#import "config.h"
+#import "BarcodeDetectorImplementation.h"
 
-#include "BarcodeFormatInterface.h"
-#include "DetectedBarcodeInterface.h"
+#if HAVE(SHAPE_DETECTION_API_IMPLEMENTATION)
+
+#import "BarcodeDetectorOptionsInterface.h"
+#import "BarcodeFormatInterface.h"
+#import "DetectedBarcodeInterface.h"
+#import "ImageBuffer.h"
+#import "NativeImage.h"
+#import "VisionUtilities.h"
+#import <Vision/Vision.h>
+#import <optional>
+#import <wtf/RetainPtr.h>
 
 namespace WebCore::ShapeDetection {
 
-BarcodeDetectorImpl::BarcodeDetectorImpl(const BarcodeDetectorOptions&)
+static BarcodeFormat convertSymbology(VNBarcodeSymbology symbology)
+{
+    if (symbology == VNBarcodeSymbologyAztec)
```

Comment:
Are these guaranteed to be tagged pointers? Otherwise we should use `isEqualToString` or `isEqual` I think.

But it would be preferable to use direct comparison, it avoids many string comparisons, assuming this is guaranteed by the framework.

Reply:
That's a good point! I'll ask the Vision team.

If comparing by pointers is correct, then this could be turned into a switch statement anyway.

Reply:
Vision team says “use -isEqual”